python/eos/plot/plotter.py

```diff
@@ -374,80 +361,75 @@ class Observable(BasePlot):
            }
         """)
 
+        from .specs import Specs, ParametersSpec, KinematicsSpec,\
+        RangeSpec, ObservableSpec, VariableSpec, ParameterFileSpec
+
+        specs_prototype = Specs([
+            ObservableSpec(name='observable'),
+            VariableSpec(name='variable'),
+            KinematicsSpec(name='kinematics', optional=True, default={}),
+            ParametersSpec(name='parameters', optional=True, default={}),
+            # ParameterFileSpec(name='parameter-file', optional=True, default=None)
+        ])
+
         def __init__(self, plotter, item):
             super().__init__(plotter, item)
+            self.specs = deepcopy(self.specs_prototype)
 
         def plot(self):
-            item = self.item
-            oname = item['observable']
-
-            obs_entry = eos.Observables._get_obs_entry(oname)
-            valid_kin_vars = [kv for kv in obs_entry.kinematic_variables()]
-            eos.info('   plotting EOS observable "{}"'.format(oname))
+            item = self.item.copy()
 
-            # create kinematics
             kinematics = eos.Kinematics()
-            if 'kinematics' in item:
-                for k, v in item['kinematics'].items():
-                    if k not in valid_kin_vars:
-                        raise ValueError("Kinematic quantity '" + k + "' does not " +
-                        "match known ones for observable '" + oname + "': " + valid_kin_vars.__repr__())
-                    kinematics.declare(k, v)
-
-            # create parameters
             parameters = eos.Parameters.Defaults()
-            if 'parameters' in item and 'parameters-from-file' in item:
-                eos.warn('    overriding values read from \'parameters-from-file\' with explicit values in \'parameters\'')
-
-            if 'parameters-from-file' in item and type(item['parameters-from-file']) is str:
-                eos.warn('    overriding parameters from file')
-                parameters.override_from_file(item['parameters-from-file'])
-
-            if 'parameters' in item and type(item['parameters']) is dict:
-                for key, value in item['parameters'].items():
-                    parameters.set(key, value)
-
-            # create (empty) options
             options = eos.Options()
 
+            # process user input
+            # need oname for some specs
+            try:
+                oname = item['observable']
+            except KeyError:
+                raise ValueError("Missing key: 'observable'")
```

Comment:
Shouldn't this exception been handled by `ObservableSpec` l.368?

Reply:
Yes, I don't like it either. But we have to obtain the oname for the next few lines. I don't see a different solution.

At least this is safe: if `observable_name` is not set before validation, an `ImplementationError` is raised, which will be caught by the tests.

Reply:
Hm, can we pass the ``ObservableSpec`` into those other specs that require the name? As a guaranteed variable?

Reply:
I think a good approach is to pass a reference of `Specs` (let's rename that to `SpecCollection`) to each owned Spec.
Then, they can access the other members and it's very easy to have dependencies. For example, making the Specs/keys 'parameters' and 'parameters-file' aware of each other. All of that without tightly coupling the `Specs` themselves.